packages/typespec-go/src/tcgcadapter/types.ts

```diff
@@ -610,13 +610,27 @@ export class typeAdapter {
       if (valueType.enumType.access === 'internal') {
         valueTypeName = naming.getEscapedReservedName(uncapitalize(valueTypeName), 'Type');
       }
-      let value = this.constValues.get(valueTypeName);
-      if (!value) {
-        value = new go.ConstantValue(valueTypeName, type, valueType.value);
-        value.docs.summary = valueType.summary;
-        value.docs.description = valueType.doc;
-        this.constValues.set(valueTypeName, value);
-      }
+
+      if (this.constValues.has(valueTypeName)) {
+        // we have a type name collision. while tsp will prevent this
+        // from directly being authored, it can happen due to how names
+        // are computed. consider the following example.
+        //
+        // union ContentTypes { "application/json" | "application/*+json" }
```

Comment:
We could consider replacing a `*` with the moniker `Wildcard` which would also resolve the issue (we'd still test for collisions and error out of course).

Reply:
Will leave the behavior as-is for now.